### PR TITLE
[POAE7-2624] support is distinct from for nextgen

### PR DESCRIPTION
**Pull request by yma11**
### What changes were proposed in this pull request?
support IS DISTINCT FROM codegen

### Why are the changes needed?
support IS DISTINCT FROM codegen

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

File: cider/type/plan/BinaryExpr.cpp [hunk 1]
```diff
@@ -121,8 +122,56 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
...
+                 lhs_val.getValue() != rhs_val.getValue()) ||
+                (lhs_val.getNull() != rhs_val.getNull());
+      } else if (lhs_val.isNullable()) {
+        value = lhs_val.getValue() != rhs_val.getValue() || lhs_val.getNull();
+      } else if (rhs_val.isNullable()) {
+        value = lhs_val.getValue() != rhs_val.getValue() || rhs_val.getNull();
+      }
+      return set_expr_value(func.createConstant(JITTypeTag::BOOL, false), value);
+    }
+    case kBW_EQ: {
```
Comment (bigPYJ1151) [inline cider/type/plan/BinaryExpr.cpp:158]: Seems this implementation could be simplified. For example, result of ```kBW_EQ``` could be calculated from the ```NOT``` of the result of ```kBW_NE```.

Comment (yma11) [reply cider/type/plan/BinaryExpr.cpp:145]: The original implementation of `lower_bw_eq` and `lower_bw_ne` has separate logic. But in semantic, we can leverage the `NOT` op. Updated it and will remove the nullable check after [PR](https://github.com/intel/BDTK/pull/218/) merged which should be more straight forward.

File: cider/type/plan/BinaryExpr.cpp [hunk 2]
```diff
@@ -121,8 +122,56 @@ JITExprValue& BinOper::codegenFixedSizeColCmpFun(JITValuePointer& null,
...
   }
+}
 
-  return expr_var_;
+JITExprValue& BinOper::codegenFixedSizeLogicalFun(JITValuePointer& null,
+                                                  JITValue& lhs,
+                                                  JITValue& rhs) {
+  switch (get_optype()) {
+    case kAND:
+      return set_expr_value(null, lhs && rhs);
```
[the remark this answers is not in the document]
Comment (yma11) [reply cider/type/plan/BinaryExpr.cpp]: Yes. You are correct. This method is actually not triggered and I will submit a new PR for it with UT covered.

**Comment by yma11** [comment]
@bigPYJ1151 @jikunshang @kd97 please help review this PR. Thanks.

**Review by bigPYJ1151 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**